test/powershell/Language/Scripting/CommandErrorHandling.Tests.ps1

```diff
@@ -0,0 +1,470 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe 'Command error handling in general' -Tag 'CI' {
+
+    BeforeAll {
+        $pwsh = [PowerShell]::Create()
+        $pwsh.AddScript(@'
+
+    function ThrowTerminatingError {
+        [CmdletBinding()]
+        param()
+
+        $ex = [System.ArgumentException]::new('terminating-exception')
+        $er = [System.Management.Automation.ErrorRecord]::new($ex, 'ThrowTerminatingError:error', 'InvalidArgument', $null)
+        $PSCmdlet.ThrowTerminatingError($er)
+
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+    function ErrorActionStop {
+        [CmdletBinding()]
+        param()
+
+        Get-Command NonExist -ErrorAction Stop
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+    function ThrowException {
+        [CmdletBinding()]
+        param()
+
+        throw 'throw-exception'
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+    function WriteErrorAPI {
+        [CmdletBinding()]
+        param()
+
+        $ex = [System.ArgumentException]::new('arg-exception')
+        $er = [System.Management.Automation.ErrorRecord]::new($ex, 'WriteErrorAPI:error', 'InvalidArgument', $null)
+        $PSCmdlet.WriteError($er)
+
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+    function WriteErrorCmdlet {
+        [CmdletBinding()]
+        param()
+
+        Write-Error 'write-error-cmdlet'
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+    function MethodInvocationThrowException {
+        [CmdletBinding()]
+        param()
+
+        ## This method call throws exception.
+        $iss = [initialsessionstate]::Create()
+        $iss.ImportPSModule($null)
+
+        Write-Verbose -Verbose "verbose-message"
+    }
+
+    function ExpressionThrowException {
+        [CmdletBinding()]
+        param()
+
+        1/0 ## throw exception.
+        Write-Verbose -Verbose "verbose-message"
+    }
+
```

Comment:
Another good one here is parameter binding, where I hit some strangeness in https://github.com/PowerShell/PowerShell/pull/15833

Reply:
It would be great if you can add tests here (or to another file as you see fit) to cover the scenario after you get it fixed in #15833.